src/apify/storage_clients/_apify/_request_queue_client.py

```diff
@@ -242,17 +242,66 @@ async def add_batch_of_requests(
         Returns:
             Response containing information about the added requests.
         """
-        # Prepare requests for API by converting to dictionaries.
-        requests_dict = [
-            request.model_dump(
-                by_alias=True,
-                exclude={'id'},  # Exclude ID fields from requests since the API doesn't accept them.
-            )
-            for request in requests
-        ]
+        # Do not try to add previously added requests to avoid pointless expensive calls to API
+
+        new_requests: list[Request] = []
+        already_present_requests: list[dict[str, str | bool]] = []
```

Comment:
This type is technically correct, but not very helpful. Since we run it through validation afterwards, perhaps we could just do `dict[str, Any]`?